wasm-engine/tests/web.rs

```diff
@@ -6,45 +6,51 @@ extern crate wasm_bindgen_test;
 use serde_wasm_bindgen::to_value;
 use wasm_bindgen::prelude::*;
 use wasm_bindgen_test::*;
-use yggdrasil_engine::{evaluate, Context};
+use yggdrasil_engine::Engine;
+use unleash_types::client_features::{ClientFeature, ClientFeatures, Constraint, Context, Operator, Strategy};
 
 #[wasm_bindgen_test]
-fn dsl_evaluates_correct() {
-    let dsl_fragment = "user_id > 5";
+fn is_enabled_evaluates_correctly() {
+    let mut engine = Engine::new();
+
+    let state: JsValue = to_value(&ClientFeatures {
+        version: 2,
+        features: vec![ClientFeature {
+            name: "feature".into(),
+            enabled: true,
+            strategies: Some(vec![Strategy {
+              name: "default".into(),
+              constraints: Some(vec![Constraint {
+                  context_name: "userId".into(),
+                  operator: Operator::In,
+                  values: Some(vec!["5".into()]),
+                  case_insensitive: false,
+                  inverted: false,
+                  value: None,
+              }]),
+              segments: None,
+              variants: None,
+              parameters: None,
+              sort_order: None,
+            }]),
+            ..ClientFeature::default()
+        }],
+        segments: None,
+        query: None,
+    }).unwrap();
+
+    let _ = engine.take_state(state);
 
     let context: JsValue = to_value(&Context {
-        user_id: Some("6".into()),
+        user_id: Some("5".into()),
         session_id: None,
         environment: None,
         app_name: None,
         current_time: None,
         remote_address: None,
-        group_id: None,
         properties: None,
-    })
-    .unwrap();
+    }).unwrap();
 
-    let result = evaluate(dsl_fragment, context).unwrap();
+    let result = engine.is_enabled("feature", context).unwrap();
     assert_eq!(result, true);
 }
```

Comment:
@sighphyre I need some insights regarding the original goal of this file. The comment at the top mentions:

> //! Test suite for the Web and headless browsers.

How so? The way I see it the only thing this test was doing was testing DSL evaluation, which I don't think belongs here.

Should we just yeet this for now?

Reply:
Ye, we can yeet it. 

These are wasm-bindgen tests, they're basically a quick and cheap way to check that your Rust code is converted to WASM correctly and executes the way you expect. The browser is only relevant insofar as its a valid WASM engine. Moving all the testing to typescript seems fine though, keeps it all in one place and removes the need for the wasm bindgen test harness which is nice

Reply:
Yep, yeeted in https://github.com/Unleash/yggdrasil/pull/143/commits/39d8bb6fd9e79d27ef0f14ca7910ef4c921a818b